tests/app/clients/test_twilio.py

```diff
@@ -599,19 +599,43 @@ def test_send_sms_twilio_callback(
         assert response_dict['sid'] == twilio_sid
 
 
+@pytest.mark.parametrize(
+    (
+        'response_dict',
+        'exception',
+    ),
+    (
+        (
+            {
+                'code': 21606,
+                'message': "The 'From' phone number provided (+61412345678) is not a valid message-capable Twilio phone number for this destination",
+            },
+            InvalidProviderException,
+        ),
+        (
+            {
+                'code': 21617,
+                'message': 'Unable to create record: The concatenated message body exceeds the 1600 character limit',
+            },
+            NonRetryableException,
+        ),
+    ),
+    ids=(
+        'invalid-from-number',
+        'message-too-long',
+    ),
+)
 def test_send_sms_raises_invalid_provider_error_with_invalid_twilio_number(
     notify_api,
     mocker,
+    response_dict,
+    exception,
 ):
     to = '+61412345678'
     content = 'my message'
     reference = 'my reference'
-    response_dict = {
-        'code': 21606,
-        'message': "The 'From' phone number provided (+61412345678) is not a valid message-capable Twilio phone number for this destination",
-    }
 
-    with pytest.raises(InvalidProviderException):
+    with pytest.raises(exception):
```

Comment:
the result of parametrizng this is that now you are not testing for the specific exception that could be raised by either of the test cases. so, now it will accept any exception and pass the test. i'm not sure this is a good idea.

Reply:
Probably a poor variable name choice, but it's lowercase, so it's not the `Exception` class.